posthog/clickhouse/cancel.py

```diff
@@ -2,13 +2,44 @@
 
 from posthog.api.services.query import logger
 from posthog.clickhouse.client import sync_execute
+from posthog.clickhouse.client.connection import default_client
 from posthog.settings import CLICKHOUSE_CLUSTER
 
 
 def cancel_query_on_cluster(team_id: int, client_query_id: str) -> None:
-    result = sync_execute(
-        f"KILL QUERY ON CLUSTER '{CLICKHOUSE_CLUSTER}' WHERE query_id LIKE %(client_query_id)s",
-        {"client_query_id": f"{team_id}_{client_query_id}%"},
-    )
+    initiator_host = None
+    failed_to_find_initiator_host = False
+    try:
+        result = sync_execute(
+            f"""
+            SELECT hostname()
+            FROM clusterAllReplicas(posthog, system.query_log)
+            WHERE query_id LIKE %(client_query_id)s AND (event_time > (now() - toIntervalDay(1)))
+            GROUP BY hostname()
+            HAVING (count() = 1)
```

Comment:
This `count() = 1` is because I expect to only have a `QueryStart` query. If there are 2 occurrences, the query has already ended somehow (either normally or because of an exception) so we don't need to kill it.

Reply:
Do we ever reuse query_id (or allow to clickhouse to set it?)?

Reply:
The `client_query_id` is not exactly the `query_id` that ClickHouse assigns as far as I could see, but a concatenation of team_id + uuid, and I have seen that the combination can appear more than once in the query_log. The final `query_id` in ClickHouse has a third string concatenated, that's why it is used the `LIKE` here. So even the final `query_id` cannot be repeated, `client_query_id` can be.